pkg/apis/v1alpha2/aksnodeclass.go

```diff
@@ -60,6 +158,13 @@ type AKSNodeClass struct {
 	Status AKSNodeClassStatus `json:"status,omitempty"`
 }
 
+// We need to bump the AKSNodeClassHashVersion when we make an update to the AKSNodeClass CRD under these conditions:
```

Comment:
This significantly concerns me. I could easily see changing this being missed. However, I don't have a great other solution.

Need good unit tests for this to try and ensure we catch all the cases where this would need to be updated, so folks are aware, inspect the failures, and update this.

Reply:
Ack. There is a design doc around this that I will link to